src/v/iceberg/values_bytes.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/values_bytes.h"
+
+#include "bytes/iobuf_parser.h"
+
+#include <variant>
+
+namespace iceberg {
+
+namespace {
+
+struct value_to_bytes_visitor {
+    template<typename T>
+    requires(std::is_scalar_v<std::decay_t<decltype(T::val)>>)
```

Comment:
should this be is_arithmetic, or do you also need to cover enum? i think pointers are also scalars (or a full enumeration of the exactly types `T` (instead of the underlying types T::val)?

Reply:
Nah, no need for enums. Done